packages/alfa-selector/src/selector.ts

```diff
@@ -1086,6 +1091,53 @@ export namespace Selector {
     private constructor() {
       super("focus");
     }
+
+    public matches(element: Element, context: Context): boolean {
+      return context.isFocused(element);
+    }
+  }
+
+  /**
+   * @see https://drafts.csswg.org/selectors/#focus-within-pseudo
+   */
+  export class FocusWithin extends Pseudo.Class {
+    public static of(): FocusWithin {
+      return new FocusWithin();
+    }
+
+    private constructor() {
+      super("focus-within");
+    }
+
+    public matches(element: Element, context: Context): boolean {
+      return (
+        context.isFocused(element) ||
+        element
```

Comment:
Don't we have an `inclusiveDescendants` method?

Reply:
We sure do!